### PR TITLE
Allow passing file access property lists when opening files

### PR DESCRIPTION
**Pull request by eschnett**
- Add keyword arguments for file create and file access property lists
- Add function to set and get libver_bounds file access property
- Add function to hide HDF5's error output (useful if a function call may fail)
- Hide errors when closing property lists in a finalizer, since the file may already have been closed

The HDF5 package seems to have a general design problem in that finalizers may close HDF5 objects that do not exist any more if the respective file has already been closed. This leads to errors. Hiding the respective HDF5 errors is straightforward (and harmless); however, there are also libuv errors that I don't know how to suppress. These look like
```
jl_uv_writecb() ERROR: bad file descriptor EBADF
```
They are harmless.

### CODE REVIEW

**Comment by timholy** [comment]
With regards to the double-close problem, it's been a long time...but I thought we added a `toclose` field to most objects. I guess the issue is that `CLOSE_STRONG` doesn't set that to `false`?

I haven't checked the code closely, but on a brief scan it looks fine. Thanks!

I won't merge yet in hopes someone can comment about the libuv errors.

**Comment by eschnett** [comment]
I think the errors are caused by `HDF5Property`; I know they caused HDF5 error output. Since this is a finalizer, it's really difficult to find out what actually causes it. `HDF5Property` does not have a `toclose` field. However -- how would a property know that a respective file has been closed?

With respect to the API, I just realized that other high-level functions don't take property lists as arguments, but rather take pairs of arguments where the first is a string and the other a tuple, and are presumably constructing properties from that. This is indeed easier to use, since much of the lower-level property functionality is not exported. I guess this should be changed before this is merged.

**Comment by eschnett** [comment]
I don't know what's wrong with coveralls. Apart from this, the patch is now ready to merge: The libuv errors are gone, and I updated the interface to take keyword/value pairs to set properties, like the other highlevel routines.

File: src/plain.jl [hunk 1]
```diff
@@ -535,31 +538,55 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool)
+function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool
+, ff::Bool,
```
Comment (timholy) [inline src/plain.jl]: Line formatting seems weird here.

**Comment by timholy** [comment]
LGTM. Just hit merge when you're ready.

**Comment by eschnett** [comment]
@timholy There's no merge button for me. Probably lack of write permissions.

**Pull request merged**